pilot/info/jobdata.py

```diff
@@ -614,3 +619,53 @@ def get_max_workdir_size(self):
             logger.warning("found no stored workdir sizes")
 
         return maxdirsize
+
+    def get_job_option_for_input_name(self, input_name):
+        """
+        Expecting something like --inputHitsFile=@input_name in jobparams.
+
+        :returns: job_option such as --inputHitsFile
+        """
+        job_options = self.jobparams.split(' ')
+        input_name_option = '=@%s' % input_name
+        for job_option in job_options:
+            if input_name_option in job_option:
+                return job_option.split("=")[0]
+        return None
+
+    def process_writetofile(self):
```

Comment:
It looks like that this actually should be a clean function like `clean__writetofile`?

Reply:
Here is the situation: AthenaMP joboption cannot support very long parameters. For esmerge jobs, there are 1000 input files and the joboption is very long. So athenaMP requires to create a txt file as a job option and put all input files in this txt file.

This function needs to create a text file in the job's working directory. In job parsing step, job's working directory is not set or maybe will be changed later. So I didn't put it in clean function. This one is called before starting AthenaMP. At that time, the job's working directory should not be changed.